extensions/amp-autocomplete/0.1/amp-autocomplete.js

```diff
@@ -572,12 +581,30 @@ export class AmpAutocomplete extends AMP.BaseElement {
     // Toggle results.
     return this.mutateElement(() => {
       if (!display) {
+        this.userInput_ = this.inputElement_.value;
+        this.filterDataAndRenderResults_(this.sourceData_, this.userInput_);
         this.resetActiveElement_();
+      } else {
+        this.setResultDisplayDirection_();
       }
       this.toggleResults_(display);
     });
   }
 
+  /**
+   * Display results upwards or downwards based on location in the viewport.
+   * @private
+   */
+  setResultDisplayDirection_() {
+    const viewHeight = this.viewport_.getHeight() || 0;
+    if (this.inputElement_.getBoundingClientRect().top > (viewHeight / 2)) {
```

Comment:
nit: you can simplify this by saving the `if` condition in a boolean and using that value as the second argument to `classList.toggle`

e.g.

```js
const renderAbove = this.inputElement_.getBoundingClientRect().top > (viewHeight / 2);
this.container_.classList.toggle('i-amphtml-autocomplete-results-up', renderAbove);
```